app/assets/javascripts/components/ppm-estimator.js

```diff
@@ -0,0 +1,157 @@
+(function(window, $) {
+  'use strict';
+
+  var $ppmEstimateForm = $('#ppm-estimate-form');
+  if ($ppmEstimateForm.length == 0)
+    return;
+
+  var PpmEstimator = function(options) {
+    this.$alert = options.$alert;
+    this.$form = options.$form;
+    this.$results = options.$results;
+
+    this.$form.on('submit', this.events.submit.bind(this));
+  };
+
+  PpmEstimator.prototype = {
+    events: {
+      submit: function(event) {
+        event.preventDefault();
+
+        $.ajax({
+          data: this.$form.serialize(),
+          error: this.handleAjaxError.bind(this),
+          method: this.$form.attr('method'),
+          success: this.handleAjaxSuccess.bind(this),
+          url: this.$form.attr('action')
+        });
+      }
+    },
+
+    handleAjaxError: function() {
+      this.$results.empty().attr('hidden', true);
+      this.$alert.removeAttr('hidden');
+    },
+
+    handleAjaxSuccess: function(markup) {
+      this.$alert.attr('hidden', true);
+      this.$results.html(markup).removeAttr('hidden');
+    }
+  };
+
+  new PpmEstimator({
+    $alert: $('#ppm-estimate-alert'),
+    $form: $ppmEstimateForm,
+    $results: $('#ppm-estimate-results')
+  });
+
+  var $rank = $('#rank');
+  var $weightAllowanceText = $('#weight-allowance-text');
```

Comment:
there's a lot of almost identical code for these 3 input fields. i feel like there should be a way to simplify things